### PR TITLE
util: add JSON in fieldTypeMergeRules

### PR DESCRIPTION
**Pull request by XuHuaiyu**
as the title says
refer to [this](https://github.com/mysql/mysql-server/blob/5.7/sql/field.cc/#654)
PTAL @hicqu @zz-jason

### CODE REVIEW

File: util/types/field_type.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func NewFieldType(tp byte) *FieldType {
 // Aggregation is performed by MergeFieldType function.
 func AggFieldType(tps []*FieldType) *FieldType {
 	var currType FieldType
-	for _, t := range tps {
-		if currType.Tp == mysql.TypeUnspecified {
+	for i, t := range tps {
+		if i == 0 && currType.Tp == mysql.TypeUnspecified {
```
Comment (shenli) [inline util/types/field_type.go]: Any test case covers this?

Comment (XuHuaiyu) [inline util/types/field_type.go]: yeah, the cases in the test file which I modified were wrong,
they cover this check.

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason

**Comment by hicqu** [comment]
LGTM

**Pull request merged**